setup_utils.py

```diff
@@ -0,0 +1,44 @@
+import os
+import re
+
+
+def update_pins(setup_args):
+    # Use requirements and constraints to set version pins
+    packages = set()
+    with open('./requirements.txt') as requirements:
+        for r in requirements:
+            if r.lower().strip() == 'dallinger':
+                continue
+            if not r.startswith('-') or r.startswith('#'):
+                packages.add(r.strip().lower())
+
+    requirements = []
+    REQUIREMENT_RE = re.compile(r'^(([^=]+)==[^#]+)(#.*)?$')
```

Comment:
With an all caps name I'd expect this to be at module scope

Reply:
Good catch. It was before I moved this into it's own module.